File: Collatz Conjecture/CC.cpp
#include <iostream>

using namespace std;

int CoF(int n, int w = 1)
{
    if(n == 1)
    {
        cout << "Returned to 1, try again lol. " << "Took " << w << " steps to do that" << endl;
        return 1;
    }
    else if(n%2)
    {
        cout << "Odd" << endl;
        return CoF(3*n+1, ++w);
    }
    else
    {
        cout << "Even" << endl;
        return CoF(n/2, ++w);  
    }
}

int main()
{
    int n;
    cout << "Input a positive integer: ";
    cin >> n;
    CoF(n);
    
   return 0; 
}
    

File: Collatz Conjecture/CC.py
n = int(input('\n'"Insert a number: "))
if n%2:
    print("Is odd")
else:
    print("Is even")
print("________________")
print("Starting..."'\n')
w = 0
while n != 1:
    if n%2 == 0:
        n = n//2
        print (n)
    else:
        n = 3*n+1
        print (n)
    w = w + 1
if n == 1:
        print('\n'"Returned to 1. Took", w, "steps to terminate, try again")
